File: Source/Aladdin.CAPI.STB/ASN1/STB/BDSBDHParams.cs
namespace Aladdin.ASN1.STB
{
    ///////////////////////////////////////////////////////////////////////////////
    // BDSBDHParams ::= CHOICE {
    //		bdsbdhParamsReference OBJECT IDENTIFIER,
    //		bdsbdhParamsList BDSBDHParamsList
    //	}
    ///////////////////////////////////////////////////////////////////////////////
    public class BDSBDHParams : Choice
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(), Cast.N), 
		    new ObjectInfo(new ObjectCreator<BDSBDHParamsList>().Factory(), Cast.N), 
	    }; 
	    // конструктор
	    public BDSBDHParams() : base(info) {} 
    }
}

File: Source/Aladdin.CAPI.STB.Java/src/aladdin/asn1/stb/BDSBDHParams.java
package aladdin.asn1.stb;
import aladdin.asn1.*; 

///////////////////////////////////////////////////////////////////////////////
// BDSBDHParams ::= CHOICE {
//		bdsbdhParamsReference OBJECT IDENTIFIER,
//		bdsbdhParamsList BDSBDHParamsList
//	}
///////////////////////////////////////////////////////////////////////////////
public final class BDSBDHParams extends Choice
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(BDSBDHParamsList.class).factory(), Cast.N), 
	}; 
	// конструктор
	public BDSBDHParams() { super(info); } 
}
